spec/models/broadcast_event/pool_released_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe BroadcastEvent::PoolReleased, type: :model, broadcast_event: true do
+  let(:source_plate) do
+    pl = create :full_stock_plate
+    pl.wells.first.aliquots << (create :aliquot, sample: tube.samples.first)
+    pl.wells.last.aliquots << (create :aliquot, sample: tube.samples.last)
+    tube.ancestors << pl
+    pl
+  end
+  let(:tube) { create :multiplexed_library_tube, sample_count: 2, purpose: (create :illumina_htp_mx_tube_purpose) }
+
+  let(:submission) { create :library_submission }
+  let(:order) { submission.orders.first }
+  let(:request1) do
+    create :multiplex_request,
+           asset: source_plate.wells.first,
+           target_asset: tube.receptacle,
+           state: 'passed',
+           order: order
+  end
+  let(:request2) do
+    create :multiplex_request,
+           asset: source_plate.wells.last,
+           target_asset: tube.receptacle,
+           state: 'passed',
+           order: order
+  end
+  let(:library_request) { create :library_request, target_asset: source_plate.wells.first }
+
+  let(:event) { described_class.create!(seed: tube, user: create(:user), properties: { order_id: order.id }) }
+  let(:subject_hash) { event.as_json['event'][:subjects].group_by(&:role_type) }
+  let(:metadata) { event.as_json['event'][:metadata] }
+
+  before do
+    request1
+    request2
+    library_request
+  end
```

Comment:
If you defined this variables with let! it creates them immediately code in that part of code, however I've seen other places where this before{} is used because it doesn work exactly that way. Did you try let!() ?

Reply:
I didn't try let!, I considered it but I found this more explicit 🤷‍♀️ 
let! would probably work too